Mathlib/Data/Bitvec/Defs.lean

```diff
@@ -1,339 +1,143 @@
 /-
-Copyright (c) 2015 Joe Hendrix. All rights reserved.
+Copyright (c) 2023 Wojciech Nawrocki. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Joe Hendrix, Sebastian Ullrich
+Authors: Harun Khan, Alex Keizer, Abdalrhman M Mohamed
 -/
-import Mathlib.Data.Vector.Basic
-import Mathlib.Data.Nat.Pow
-import Init.Data.Format.Basic
-import Mathlib.Init.Data.Nat.Lemmas
+
+import Mathlib.Data.Fin.Basic
+import Mathlib.Data.Nat.Bitwise
+import Mathlib.Data.ZMod.Defs
+import Std.Data.BitVec
 
 #align_import data.bitvec.core from "leanprover-community/mathlib"@"1126441d6bccf98c81214a0780c73d499f6721fe"
+
 /-!
 # Basic operations on bitvectors
 
-This is a work-in-progress, and contains additions to other theories.
-
-This file was moved to mathlib from core Lean in the switch to Lean 3.20.0c.
-It is not fully in compliance with mathlib style standards.
--/
-
-/-- `Bitvec n` is a `Vector` of `Bool` with length `n`. -/
-@[reducible]
-def Bitvec (n : ℕ) :=
-  Vector Bool n
-#align bitvec Bitvec
-
-namespace Bitvec
-
-open Nat
+Std has defined bitvector of length `w` as `Fin (2^w)`.
+Here we define a few more operations on these bitvectors
 
-open Vector
+## Main definitions
 
--- mathport name: «expr ++ₜ »
-local infixl:65 "++ₜ" => Vector.append
+* `Std.BitVec.sgt`: Signed greater-than comparison of bitvectors
+* `Std.BitVec.sge`: Signed greater-equals comparison of bitvectors
+* `Std.BitVec.ugt`: Unsigned greater-than comparison of bitvectors
+* `Std.BitVec.uge`: Unsigned greater-equals comparison of bitvectors
+* `Std.BitVec.ofLEList`: Turn a little-endian list of Booleans into a bitvector
+* `Std.BitVec.ofBEList`: Turn a big-endian list of Booleans into a bitvector
```

Comment:
I don't see these below in the file.

Reply:
These were referring functions that we decided to not upstream in the initial PR. They are now removed.